net/DevExtreme.AspNet.Data.Tests/GroupHelperTests.cs

```diff
@@ -127,6 +127,29 @@ public class GroupHelperTests {
         }
 
         [Fact]
+        public void GroupInterval_NullDates() {
+            var data = new[] {
+                new { d = new DateTime?() },
+                new { d = new DateTime?() }
+            };
+
+            var groups = CreateHelper(data).Group(data, new[] {
+                new GroupingInfo { Selector = "d", GroupInterval = "year", IsExpanded = false },
+                new GroupingInfo { Selector = "d", GroupInterval = "month", IsExpanded = false },
+                new GroupingInfo { Selector = "d", GroupInterval = "day", IsExpanded = false }
+            });
+
+            var g_year = groups[0];
+            var g_month = g_year.items[0] as Group;
+            var g_day = g_month.items[0] as Group;
+
+            Assert.Equal(groups.Count, 1);
```

Comment:
Well, then you need to check count on all 3 levels 😉
And do it prior to using indexers.

This issue being fixed is about wrong group keys, not about wrong item counts.